File: app/src/main/java/ru/kuchanov/scpquiz/api/QuizApi.java
package ru.kuchanov.scpquiz.api;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;
import ru.kuchanov.scpquiz.api.response.TokenResponse;
import ru.kuchanov.scpquiz.model.api.NwQuiz;

public interface QuizApi {
    @FormUrlEncoded
    @POST("scp-quiz/oauth/token")
    Observable<TokenResponse> getAccessToken(
            @Header("Authorization") String authorization,
            @Field("grant_type") String grantType
    );


    @GET("scp-quiz/quiz/all")
    Observable<List<NwQuiz>> getNwQuizList(
            @Header("Authorization") String authorization
    );
}

File: app/src/main/java/com/scp/scpexam/controller/api/QuizApi.kt
package com.scp.scpexam.controller.api

import io.reactivex.Single
import retrofit2.http.*
import com.scp.scpexam.controller.api.response.TokenResponse
import com.scp.scpexam.model.api.NwQuiz
import io.reactivex.Flowable

interface QuizApi {

    /**
     * use it only for get quizList if user doesn't log in.
     */
    @FormUrlEncoded
    @POST("oauth/token")
    fun getUnloginedUserAccessToken(
            @Header("Authorization") authorization: String,
            @Field("grant_type") grantType: String
    ): Single<TokenResponse>

    /**
     * use token from [getUnloginedUserAccessToken]
     */
    @GET("quiz/all")
    fun getNwQuizList(): Single<List<NwQuiz>>

    @GET("quiz/all/pagination")
    fun getNwQuizListPaging(
            @Query("offset") offset: Int,
            @Query("limit") limit: Int
    ): Flowable<List<NwQuiz>>

}

